lib/jekyll/renderer.rb

```diff
@@ -234,6 +235,14 @@ def assign_highlighter_options!
     end
 
     private
+    def assign_layout_data!
+      layout = layouts[document.data["layout"]]
```

Comment:
Not sure... basically the problem is that we don't have layout data in `payload` when we render document [here](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/renderer.rb#L75). I've tried to move the  `place_in_layouts` call before rendering the document (because we add layout data to `payload` in that method), but that breaks a lot of tests.

Not sure how to fix this without too much changes :)